### ISSUE TITLE
Distributed mode overwrites the user's choice for dataloaders shuffling

### ISSUE THREAD

**Issue by eladsegal** [PyTorchLightning/pytorch-lightning#10170]
## 🐛 Bug
I ordered my training data in a specific manner and passed it to the DataLoader with `shuffle=False` (I use `reload_dataloaders_every_n_epochs=1` to control it every epoch). Then, I found out that running in distributed mode re-creates the train dataloader and always set `shuffle=True`. The opposite happens for the val dataloader.

I think that the re-creation of the dataloaders must be using the same shuffle setting the original dataloaders had.
Is there any reason not to do so?

---
Here is the related code with the hard-coded shuffle values:  

train - shuffle forced to be true
https://github.com/PyTorchLightning/pytorch-lightning/blob/c33df2639f19d49c5e7520294e3221efe402d684/pytorch_lightning/trainer/data_loading.py#L330-L333

val - shuffle forced to be true
https://github.com/PyTorchLightning/pytorch-lightning/blob/c33df2639f19d49c5e7520294e3221efe402d684/pytorch_lightning/trainer/data_loading.py#L442-L443

cc @borda @justusschock @awaelchli

**Comment by arielsho** [comment]
I came across the same problem when I was trying to debug my code today.

**Comment by francescocarzaniga** [comment]
I was a bit stumped by this today, as I didn't really find any indication of this behaviour in the docs. In my case I am already shuffling the dataset (albeit in a particular way), and shuffling again is both needlessly expensive and causes the wrong behaviour.
In general I believe most people are using `RandomShuffler`, so shuffling again is really not useful. For this reason IMHO shuffle in `DistributedSampler` should be off by default, with a warning (like for test and val with `shuffle=True`) and possibly a flag.
